### DISCUSSION TITLE
谢老师您好我今天上午参加你在华中农业大学的报告研究生

### DISCUSSION THREAD

**Discussion by wangguiqian** [rbind/yihui#1497] [Q&A]
您好！

谢老师请问，我主要是运用R语言的，主要是通过VS-code的上面的jupyter来调用服务器上面的R但是很多时候安装R包的时候例如：“seurat”时候显示no-zero   ，但是在我本地电脑的R上面安装可以正常运行请问这个是什么原因？

祝：身体健康、工作顺利！

**Comment by yihui** [comment]
出现 non-zero status 的可能性不唯一，所以很难讲，比如文件夹不可写、代码编译失败、缺少系统依赖，等等。没有更多错误信息了吗？

**Comment by wangguiqian** [comment]
<img width="565" alt="image" src="https://github.com/yihui/yihui.org/assets/48958211/ea826c6d-d8c1-447a-956a-c090fdb815e6">

谢老师你看我有时候安装包的时候就是经常出现这样问题

**Reply by yihui** [reply]
有些包是旧版本的 R 下面安装的，你升级 R 之后需要重装它们。

```r
update.packages(ask = FALSE, checkBuilt = TRUE)
```

又及：能复制文本就复制文本粘贴出来，不需要截图。

**Comment by wangguiqian** [comment]
<img width="860" alt="image" src="https://github.com/yihui/yihui.org/assets/48958211/0420927d-1c66-47b8-998b-ca1b5ee5ba63">
例如当我再次安装这个包的时候就也会出现这样的问题，但是在我本地的电脑上的R的进行安装就没有问题，这个全部在通过VS-code远程连接Linux服务器使用R时出现问题

**Comment by wangguiqian** [comment]
<img width="957" alt="image" src="https://github.com/yihui/yihui.org/assets/48958211/ea596030-b6f8-4119-8109-41fe39f96d19">

**Comment by wangguiqian** [comment]
**update.packages(ask = FALSE,checkBuilt = TRUE)**


Warning message in install.packages(update[instlib == l, "Package"], l, contriburl = contriburl, :
“installation of package ‘cluster’ had non-zero exit status”
Warning message in install.packages(update[instlib == l, "Package"], l, contriburl = contriburl, :
“installation of package ‘collections’ had non-zero exit status”
Warning message in install.packages(update[instlib == l, "Package"], l, contriburl = contriburl, :
“installation of package ‘commonmark’ had non-zero exit status”
Warning message in install.packages(update[instlib == l, "Package"], l, contriburl = contriburl, :
“installation of package ‘curl’ had non-zero exit status”
Warning message in install.packages(update[instlib == l, "Package"], l, contriburl = contriburl, :
“installation of package ‘digest’ had non-zero exit status”
Warning message in install.packages(update[instlib == l, "Package"], l, contriburl = contriburl, :
“installation of package ‘fansi’ had non-zero exit status”
Warning message in install.packages(update[instlib == l, "Package"], l, contriburl = contriburl, :
“installation of package ‘fastmap’ had non-zero exit status”
Warning message in install.packages(update[instlib == l, "Package"], l, contriburl = contriburl, :
“installation of package ‘foreign’ had non-zero exit status”
Warning message in install.packages(update[instlib == l, "Package"], l, contriburl = contriburl, :
“installation of package ‘fs’ had non-zero exit status”
Warning message in install.packages(update[instlib == l, "Package"], l, contriburl = contriburl, :
“installation of package ‘jsonlite’ had non-zero exit status”
Warning message in install.packages(update[instlib == l, "Package"], l, contriburl = contriburl, :
“installation of package ‘KernSmooth’ had non-zero exit status”
Warning message in install.packages(update[instlib == l, "Package"], l, contriburl = contriburl, :
“installation of package ‘lattice’ had non-zero exit status”
Warning message in install.packages(update[instlib == l, "Package"], l, contriburl = contriburl, :
“installation of package ‘lazyeval’ had non-zero exit status”
Warning message in install.packages(update[instlib == l, "Package"], l, contriburl = contriburl, :
“installation of package ‘lpSolve’ had non-zero exit status”
Warning message in install.packages(update[instlib == l, "Package"], l, contriburl = contriburl, :
“installation of package ‘MASS’ had non-zero exit status”
Warning message in install.packages(update[instlib == l, "Package"], l, contriburl = contriburl, :
“installation of package ‘nnet’ had non-zero exit status”
Warning message in install.packages(update[instlib == l, "Package"], l, contriburl = contriburl, :
“installation of package ‘pbdZMQ’ had non-zero exit status”
Warning message in install.packages(update[instlib == l, "Package"], l, contriburl = contriburl, :
“installation of package ‘png’ had non-zero exit status”
Warning message in install.packages(update[instlib == l, "Package"], l, contriburl = contriburl, :
“installation of package ‘ps’ had non-zero exit status”
Warning message in install.packages(update[instlib == l, "Package"], l, contriburl = contriburl, :
“installation of package ‘Rcpp’ had non-zero exit status”
Warning message in install.packages(update[instlib == l, "Package"], l, contriburl = contriburl, :
“installation of package ‘rpart’ had non-zero exit status”
Warning message in install.packages(update[instlib == l, "Package"], l, contriburl = contriburl, :
“installation of package ‘spatial’ had non-zero exit status”
Warning message in install.packages(update[instlib == l, "Package"], l, contriburl = contriburl, :
“installation of package ‘stringi’ had non-zero exit status”
Warning message in install.packages(update[instlib == l, "Package"], l, contriburl = contriburl, :
“installation of package ‘utf8’ had non-zero exit status”
Warning message in install.packages(update[instlib == l, "Package"], l, contriburl = contriburl, :
“installation of package ‘vctrs’ had non-zero exit status”
Warning message in install.packages(update[instlib == l, "Package"], l, contriburl = contriburl, :
“installation of package ‘xfun’ had non-zero exit status”
Warning message in install.packages(update[instlib == l, "Package"], l, contriburl = contriburl, :
“installation of package ‘xml2’ had non-zero exit status”
Warning message in install.packages(update[instlib == l, "Package"], l, contriburl = contriburl, :
“installation of package ‘yaml’ had non-zero exit status”
Warning message in install.packages(update[instlib == l, "Package"], l, contriburl = contriburl, :
“installation of package ‘zip’ had non-zero exit status”
Warning message in install.packages(update[instlib == l, "Package"], l, contriburl = contriburl, :
“installation of package ‘class’ had non-zero exit status”
Warning message in install.packages(update[instlib == l, "Package"], l, contriburl = contriburl, :
“installation of package ‘tibble’ had non-zero exit status”
Warning message in install.packages(update[instlib == l, "Package"], l, contriburl = contriburl, :
“installation of package ‘lintr’ had non-zero exit status”
Warning message in install.packages(update[instlib == l, "Package"], l, contriburl = contriburl, :
“installation of package ‘pkgdepends’ had non-zero exit status”
Warning message in install.packages(update[instlib == l, "Package"], l, contriburl = contriburl, :
“installation of package ‘readxl’ had non-zero exit status”
Warning message in install.packages(update[instlib == l, "Package"], l, contriburl = contriburl, :
“installation of package ‘roxygen2’ had non-zero exit status”
Warning message in install.packages(update[instlib == l, "Package"], l, contriburl = contriburl, :
“installation of package ‘styler’ had non-zero exit status”
Updating HTML index of packages in '.Library'

Making 'packages.html' ...
 done

**Reply by yihui** [reply]
我不知道为什么所有的包都安装失败，你可以尝试只安装其中一个，看有没有更多错误信息，比如 `install.packages('commonmark')`。

**Comment by wangguiqian** [comment]
**install.packages("commonmark")**

Warning message in install.packages("commonmark"):
“installation of package ‘commonmark’ had non-zero exit status”
Updating HTML index of packages in '.Library'

Making 'packages.html' ...
 done

请问谢老师为什么会出现这个问题我使用本地电脑的R没有问题，但是使用远程服务器的R时总会出现这个问题

**Comment by wangguiqian** [comment]
**BiocManager::install(checkBuilt=TRUE)**

Warning message:
“unable to access index for repository https://mirrors.tuna.tsinghua.edu.cn/bioconductor/packages/3.12/bioc/src/contrib:
  cannot open URL 'https://mirrors.tuna.tsinghua.edu.cn/bioconductor/packages/3.12/bioc/src/contrib/PACKAGES'”
Warning message:
“unable to access index for repository https://mirrors.tuna.tsinghua.edu.cn/bioconductor/packages/3.12/data/annotation/src/contrib:
  cannot open URL 'https://mirrors.tuna.tsinghua.edu.cn/bioconductor/packages/3.12/data/annotation/src/contrib/PACKAGES'”
Warning message:
“unable to access index for repository https://mirrors.tuna.tsinghua.edu.cn/bioconductor/packages/3.12/data/experiment/src/contrib:
  cannot open URL 'https://mirrors.tuna.tsinghua.edu.cn/bioconductor/packages/3.12/data/experiment/src/contrib/PACKAGES'”
Warning message:
“unable to access index for repository https://mirrors.tuna.tsinghua.edu.cn/bioconductor/packages/3.12/workflows/src/contrib:
  cannot open URL 'https://mirrors.tuna.tsinghua.edu.cn/bioconductor/packages/3.12/workflows/src/contrib/PACKAGES'”
Bioconductor version 3.12 (BiocManager 1.30.21), R 4.0.2 (2020-06-22)

Installing package(s) 'BiocVersion'

Warning message:
“unable to access index for repository https://mirrors.tuna.tsinghua.edu.cn/bioconductor/packages/3.12/bioc/src/contrib:
  cannot open URL 'https://mirrors.tuna.tsinghua.edu.cn/bioconductor/packages/3.12/bioc/src/contrib/PACKAGES'”
Warning message:
“unable to access index for repository https://mirrors.tuna.tsinghua.edu.cn/bioconductor/packages/3.12/data/annotation/src/contrib:
  cannot open URL 'https://mirrors.tuna.tsinghua.edu.cn/bioconductor/packages/3.12/data/annotation/src/contrib/PACKAGES'”
Warning message:
“unable to access index for repository https://mirrors.tuna.tsinghua.edu.cn/bioconductor/packages/3.12/data/experiment/src/contrib:
  cannot open URL 'https://mirrors.tuna.tsinghua.edu.cn/bioconductor/packages/3.12/data/experiment/src/contrib/PACKAGES'”
Warning message:
“unable to access index for repository https://mirrors.tuna.tsinghua.edu.cn/bioconductor/packages/3.12/workflows/src/contrib:
  cannot open URL 'https://mirrors.tuna.tsinghua.edu.cn/bioconductor/packages/3.12/workflows/src/contrib/PACKAGES'”
Warning message:
“package ‘BiocVersion’ is not available (for R version 4.0.2)”
Warning message:
“unable to access index for repository https://mirrors.tuna.tsinghua.edu.cn/bioconductor/packages/3.12/bioc/src/contrib:
  cannot open URL 'https://mirrors.tuna.tsinghua.edu.cn/bioconductor/packages/3.12/bioc/src/contrib/PACKAGES'”
Warning message:
“unable to access index for repository https://mirrors.tuna.tsinghua.edu.cn/bioconductor/packages/3.12/data/annotation/src/contrib:
  cannot open URL 'https://mirrors.tuna.tsinghua.edu.cn/bioconductor/packages/3.12/data/annotation/src/contrib/PACKAGES'”
Warning message:
“unable to access index for repository https://mirrors.tuna.tsinghua.edu.cn/bioconductor/packages/3.12/data/experiment/src/contrib:
  cannot open URL 'https://mirrors.tuna.tsinghua.edu.cn/bioconductor/packages/3.12/data/experiment/src/contrib/PACKAGES'”
Warning message:
“unable to access index for repository https://mirrors.tuna.tsinghua.edu.cn/bioconductor/packages/3.12/workflows/src/contrib:
  cannot open URL 'https://mirrors.tuna.tsinghua.edu.cn/bioconductor/packages/3.12/workflows/src/contrib/PACKAGES'”
Old packages: 'class', 'htmltools', 'knitr', 'lintr', 'Matrix', 'mgcv', 'nlme',
  'pillar', 'pkgcache', 'pkgdepends', 'processx', 'purrr', 'readxl',
  'roxygen2', 'selectr', 'stringr', 'styler', 'survival', 'tibble', 'tinytex',
  'utf8', 'vctrs', 'xfun', 'xml2', 'yaml', 'zip'

**Comment by wangguiqian** [comment]
但是我使用conda安装就没有问题,conda install r-commonmark 就可以正常安装这个R包

**Reply by yihui** [reply]
我不熟悉 Conda，它可能有它特殊的包的管理机制。

**Comment by wangguiqian** [comment]
因为我主要使用VScode的来调用R的但是目前，我没有找到这个问题的解决方法，请问谢老师是否可以明天我找老师当面帮忙解决一下呢？我是华中农业大学的研究生，今天很荣幸参加老师您的报告

**Reply by yihui** [reply]
嗯，等休息的时间我看看吧，但还是那句话，我对 Conda 不熟，不一定能帮到你。
